app/merginapi.cpp

```diff
@@ -59,14 +62,130 @@ void MerginApi::listProjects( const QString &searchExpression, const QString &us
   connect( reply, &QNetworkReply::finished, this, &MerginApi::listProjectsReplyFinished );
 }
 
+void MerginApi::downloadFile( const QString &projectFullName, const QString &filename, const QString &version, int chunkNo )
+{
+  if ( !validateAuthAndContinute() || mApiVersionStatus != MerginApiStatus::OK )
+  {
+    return;
+  }
+
+  QNetworkRequest request;
+  // chunkNo is redundant in request, but used in handling reply
+  QUrl url( mApiRoot + QStringLiteral( "/v1/project/raw/%1?file=%2&version=%3&chunkNo=%4" ).arg( projectFullName ).arg( filename ).arg( version ).arg( chunkNo ) );
+  request.setUrl( url );
+  request.setRawHeader( "Authorization", QByteArray( "Bearer " + mAuthToken ) );
+
+  QString range;
+  int from = UPLOAD_CHUNK_SIZE * chunkNo;
+  int to = UPLOAD_CHUNK_SIZE * ( chunkNo + 1 );
+  if ( chunkNo != 0 )
+    from++;
```

Comment:
this computation looks strange... if chunk size is 100, it will request 0-100, then 101-200, 201-300 ...  so the first chunk will be weirdly one byte longer (101 bytes) compared to others (100 bytes). Better to request 0-99, 100-199, ... ?